### DISCUSSION TITLE
bring Dapr service into Kiali

### DISCUSSION THREAD

**Discussion by robberphex** [kiali/kiali#4322] [Roadmap]
**Is your feature request related to a problem? Please describe.**

When using Dapr, there is no dashboard like Kiali to display service graph, app list with metric/trace.

**Describe the solution you'd like**
* In application detail, workload detail, service detail page, show Dapr traffic in traffic tab, show Dapr metrics in metrics tabs.
* At service graph, show Istio and Dapr in one graph.

**Describe alternatives you've considered**
Dapr can develop this kind of dashboard(I guess), but cannot interoperate with Istio service. For example, show Dapr service call and Istio service call at one graph.

**Additional context**

[Dapr](https://dapr.io/) is Distributed Application Runtime, which is a portable, event-driven runtime.

**Comment by robberphex** [comment]
I'm working on this way, but I'm not sure there is an alternative way or a better way.

**Comment by jmazzitelli** [comment]
Can you explain what Dapr is?

If this is asking for Kiali to support additional non-Istio service mesh implementations, that's already been requested several times in the past and declined. See:
* https://github.com/kiali/kiali/issues/2823
* https://github.com/kiali/kiali/issues/1679
* https://github.com/kiali/kiali/issues/3778

But I cannot tell if that is what this enhancement request is asking for - need details on what "Dapr" is.

**Comment by jmazzitelli** [comment]
* https://docs.dapr.io/concepts/overview/
* https://docs.dapr.io/concepts/service-mesh/

@lucasponce I wonder if there is room here to do an integration with Dapr in the same way we did with iter8?

**Comment by lucasponce** [comment]
Hi @RobberPhex I'm moving this issue into a discussionas I think it allows better to describe the user stories of your proposal.

A couple of quick answers:
- If Dapr provides some metrics, those could be added into Kiali using custom dashboards [1] 
- The topology built in Kiali is a mix of signals, mainly based on Istio traffic metrics, in a first thought this also sounds like another source of information, as it happens today with tracing, that also provides another topology and Kiali allows to correlate. 

I wonder if you could detail the use case, I mean, which information collected by Dapr would be needed to show to the user that is not collected via Istio telemetry or Tracing ?

May you provide some diagram with some example ? That would help to understand the scenario.


[1] https://kiali.io/documentation/latest/runtimes-monitoring/

**Comment by jmazzitelli** [comment]
FYI: additional details found in this Dapr issue: https://github.com/dapr/dashboard/issues/6
